src/lib/y2network/interfaces_collection.rb

```diff
@@ -65,7 +65,7 @@ def add(name)
     # @return [Boolean] true when both collections contain only equal interfaces,
     #                   false otherwise
     def ==(other)
-      @interfaces - other.interfaces && other.interfaces == @interfaces
+      ((interfaces - other.interfaces) | (other.interfaces - interfaces)).empty?
```

Comment:
well, `|` is quite expensive and complex operation. I think simple `+` is more understandable and also works.